src/averageAges.js

```diff
@@ -55,7 +74,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredWomen = people.filter((child) =>
+    onlyWithSon
+      ? people.find((women) => child.mother === women.name && child.sex === "m")
+      : people.find((women) => child.mother === women.name)
+  );
+
+  return (
+    filteredWomen.reduce(
+      (sum, child) =>
+        sum +
+        child.born -
+        people.find((women) => child.mother === women.name).born,
+      0
+    ) / filteredWomen.length
```

Comment:
prettify it plz

Reply:
And what exactly do you not like? Because it's all according to the rules, it seems)